src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1347,6 +1358,32 @@ void Transformations::MainSnippets() {
                 return is_unsupported_parallel_work_amount(n, pshape);
             },
             snippets::pass::TokenizeMHASnippets);
+        CPU_SET_CALLBACK_X64(
+            snippetsManager,
+            [&](const std::shared_ptr<const ov::Node>& n) -> bool {
+                if (!is_supported_matmul(n))
+                    return true;
+                if (is_unsupported_parallel_work_amount(n, n->get_output_partial_shape(0)))
+                    return true;
+
+                // TODO: release these conditions
+                const auto& input_shape = n->get_input_partial_shape(0);
+                if (input_shape.rank().get_length() != 2)
+                    return true;
+                if (input_shape[0].is_dynamic() || input_shape[1].is_dynamic())
+                    return true;
+                if (input_shape[0].get_length() > 8)
+                    return true;
+                if (input_shape[1].get_length() > 256)
+                    return true;
```

Comment:
Minor:
```suggestion
                if (input_shape.rank().is_dynamic() || input_shape.size() != 2)
                    return true;
                if (input_shape.is_dynamic() || input_shape[0] > 8 || input_shape[1] > 256)
                    return true;
```
For output shape the comment is the same: you can just check if shape/rank is not dynamic and then use "static" methods without any `get_length`

Reply:
Done